In [1]:
import pandas as pd
import numpy as np

In [11]:
# reading the dataset
slu_data = pd.read_csv("SLU Opportunity Wise Data.csv")

     Learner SignUp DateTime                  Opportunity Id  \
0        06/14/2023 12:30:35  00000000-0GN2-A0AY-7XK8-C5FZPP   
1        05/01/2023 05:29:16  00000000-0GN2-A0AY-7XK8-C5FZPP   
2        04/09/2023 20:35:08  00000000-0GN2-A0AY-7XK8-C5FZPP   
3        08/29/2023 05:20:03  00000000-0GN2-A0AY-7XK8-C5FZPP   
4        01/06/2023 15:26:36  00000000-0GN2-A0AY-7XK8-C5FZPP   
...                      ...                             ...   
8553     03/05/2024 16:39:21  00000000-10GQ-RJHT-3G4S-BKGBY1   
8554     03/06/2024 15:58:47  00000000-10GQ-RJHT-3G4S-BKGBY1   
8555     02/29/2024 15:02:31  00000000-10GQ-RJHT-3G4S-BKGBY1   
8556     12/23/2023 03:53:12  00000000-10GQ-RJHT-3G4S-BKGBY1   
8557     06/01/2023 13:22:01  00000000-10GQ-RJHT-3G4S-BKGBY1   

                                       Opportunity Name Opportunity Category  \
0     Career Essentials: Getting Started with Your P...               Course   
1     Career Essentials: Getting Started with Your P...               C

In [5]:
# check for the first 100 rows of the data set
slu_data.head()

Learner SignUp DateTime                  Opportunity Id  \
0     06/14/2023 12:30:35  00000000-0GN2-A0AY-7XK8-C5FZPP   
1     05/01/2023 05:29:16  00000000-0GN2-A0AY-7XK8-C5FZPP   
2     04/09/2023 20:35:08  00000000-0GN2-A0AY-7XK8-C5FZPP   
3     08/29/2023 05:20:03  00000000-0GN2-A0AY-7XK8-C5FZPP   
4     01/06/2023 15:26:36  00000000-0GN2-A0AY-7XK8-C5FZPP   

                                    Opportunity Name Opportunity Category  \
0  Career Essentials: Getting Started with Your P...               Course   
1  Career Essentials: Getting Started with Your P...               Course   
2  Career Essentials: Getting Started with Your P...               Course   
3  Career Essentials: Getting Started with Your P...               Course   
4  Career Essentials: Getting Started with Your P...               Course   

  Opportunity End Date        First Name Date of Birth  Gender        Country  \
0  06/29/2024 18:52:39             Faria    01/12/2001  Female       Pakistan   
1  06/29/2024 18:52:39          Poojitha    08/16/2000  Female          India   
2  06/29/2024 18:52:39          Emmanuel    01/27/2002    Male  United States   
3  06/29/2024 18:52:39  Amrutha Varshini    11/01/1999  Female  United States   
4  06/29/2024 18:52:39    Vinay Varshith    04/19/2000    Male  United States   

                   Institution Name Current/Intended Major  \
0                             Nwihs              Radiology   
1                       SAINT LOUIS    Information Systems   
2  Illinois Institute of Technology       Computer Science   
3            Saint Louis University    Information Systems   
4            Saint Louis University       Computer Science   

      Entry created at Status Description  Status Code  \
0  03/11/2024 12:01:41            Started         1080   
1  03/11/2024 12:01:41            Started         1080   
2  03/11/2024 12:01:41            Started         1080   
3  03/11/2024 12:01:41     Team Allocated         1070   
4  03/11/2024 12:01:41            Started         1080   

                 Apply Date Opportunity Start Date  
0       06/14/2023 12:36:09    11/03/2022 18:30:39  
1       05/01/2023 06:08:21    11/03/2022 18:30:39  
2  05/11/2023 1085640:21:29    11/03/2022 18:30:39  
3       10/09/2023 22:02:42    11/03/2022 18:30:39  
4       01/06/2023 15:40:10    11/03/2022 18:30:39

In [8]:
# check for any missing data in by columns
missing_data = slu_data.isnull().sum()
missing_data[:]

Learner SignUp DateTime       0
Opportunity Id                0
Opportunity Name              0
Opportunity Category          0
Opportunity End Date          0
First Name                    0
Date of Birth                 0
Gender                        0
Country                       0
Institution Name              5
Current/Intended Major        5
Entry created at              0
Status Description            0
Status Code                   0
Apply Date                    0
Opportunity Start Date     3794
dtype: int64

Missing data in the data set in the columns include:
1. The opportunity start date = 3794 counts
2. Institution Name = 5
3. Current/Intended Major = 5

In [10]:
total_cells = np.product(slu_data.shape)
total_missing = missing_data.sum()
percent_missing = (total_missing/total_cells) * 100
print(percent_missing)

2.778102360364571


Percentage of data missing = 2.778102360364571%